pii_recognition/registers/registry.py

```diff
@@ -1,17 +1,14 @@
-from abc import ABCMeta, abstractmethod
-from typing import TypeVar
+from typing import Generic, Optional, Type, TypeVar
 
-T = TypeVar("T")
+T_co = TypeVar("T_co", covariant=True)
 
 
-class Registry(dict, metaclass=ABCMeta):
-    def __init__(self):
-        self.add_predefines()
+class Registry(dict, Generic[T_co]):
+    def add_item(self, item: Type[T_co], name: Optional[str] = None):
+        if name:
+            self[name] = item
+        else:
+            self[getattr(item, "__name__")] = item
 
-    @abstractmethod
-    def add_predefines(self):
-        ...
-
-    def add_item(self, item: T):
-        name = getattr(item, "__name__")
-        self[name] = item
+    def create_instance(self, name: str, **config) -> T_co:
+        return self[name](**config)
```

Comment:
what does config contain?

Reply:
So far it contains `str`, `List`, `Dict`.